tests/test_fips.py

```diff
@@ -134,3 +200,101 @@ def fips_mode_setup_check(container_per_test: ContainerData) -> None:
 )
 def test_openssl_fips_hashes(container_per_test: ContainerData):
     openssl_fips_hashes_test_fnct(container_per_test)
+
+
+@pytest.mark.parametrize(
+    "container_per_test", FIPS_GNUTLS_TESTER_IMAGES, indirect=True
+)
+def test_gnutls_binary(container_per_test: ContainerData) -> None:
+    """Check that a binary linked against Gnutls obeys the host's FIPS mode
+    setting:
+
+    - build a container image using :py:const:`DOCKERFILE_GNUTLS`
+    - run the bundled binary compiled from :file:`tests/files/fips-gnutls-test.c` with
+      all FIPS digests and assert that it successfully calculates the message
+      digest
+    - rerun the same binary with non-FIPS digests and assert that this fails
+      with the expected error message.
+
+    """
+
+    expected_fips_gnutls_digests = [
+        "c87d25a09584c040f3bfc53b570199591deb10ba648a6a6ffffdaa0badb23b8baf90b6168dd16b3a",
+        "54655eae3d97147de34564572231c34d6d0917dd7852b5b93647fb4fe53ee97e5e0a2a4d359b5b461409dc44d9315afbc3b7d6bc5cd598e6",
+        "4ea6a95a3a56fa6b7c1673c145198c52265fea4fe4cebef97249b39c25a733a0d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+        "416b69644f4844065fcfe7b60f14fe07d1573420c4db2b15a6d1f4cb2b6933ffce8dcb1bce7788b962eb3d0c8d4eefc4acbfd470c22c0d95a1d10a087dc31988b9f7bfeb13be70b876a73558be664e5858d11f9459923e6e5fd838cb5708b969",
+        "3916af571551c0c40eb19936c7ac2c090e140cad48e348dc4e9d5b6508a34ac6090daeeed3a081ce0e44c90b181987b71f09e8e0c190e5af26cd46eea724489de1c112ff908febc3b98b1693a6cd3564eaf8e5e6ca629d084d9f0eba99247cacdd72e369ff8941397c2807409ff66be64be908da17ad7b8a49a2a26c0e8086aa",
+    ]
+
+    for digest, expected_digest in zip(
+        FIPS_GNUTLS_DIGESTS, expected_fips_gnutls_digests
+    ):
+        container_per_test.connection.check_output(
+            f"/bin/fips-test-gnutls {digest}"
+        )
+        res = container_per_test.connection.run_expect(
+            [0], f"/bin/fips-test-gnutls {digest}"
+        )
+        assert expected_digest in res.stdout
+
+    for digest in NONFIPS_GNUTLS_DIGESTS:
+        err_msg = container_per_test.connection.run_expect(
+            [1], f"/bin/fips-test-gnutls {digest}"
+        ).stderr
+
+        assert (
+            "Hash calculation failed" in err_msg
+        ), f"Hash calculation unexpectedly succeeded for {digest}"
+
+@pytest.mark.parametrize(
+    "container_per_test", FIPS_GCRYPT_TESTER_IMAGES, indirect=True
+)
+def test_gcrypt_binary(container_per_test: ContainerData) -> None:
```

Comment:
This needs a `pytest.mark.skipif()` for sp5

Reply:
Added skip for 15.3, 15.4, 15.5 and 15.6.
Right now libgcrypt is expected to fail on all these versions.